docs/UserGuide.md

```diff
@@ -135,6 +135,62 @@ If `NAME_STRING` = `John p/ Tom`, the command keyed in could look like `radd p/[
 
+#### Interpreting errors
+
+In most cases, the error displayed should be relatively self-explanatory. This sub-section aims to clear up any ambiguity 
+you might encounter when dealing with invalid commands or errors. 
+
+We will break our explanation down into the following categories:
+- Incorrect formats
+- Parameters do not conform to required values
+- Commands that require an [`INDEX`] (e.g `redit 1 n/John`)
+
+##### Incorrect formats
+This error typically shows a message along the following lines: `Invalid command format! [Information on how to use
+the command]`
+
+![Invalid command format](images/error-invalid-cmd-format.png)
+
+Typically this happens because the command does not match the required format. Check if you are missing any parameters.
+A special case of this takes place when the `INDEX` value is not a number.
+
+##### Parameters do not conform to required values
+This error typically shows a message along the following lines: `[Parameter name] must be [some condition]`. E.g 
+`Room numbers should be formatted as such: XY-ABC, where XY can be any pair of digits except 00, and ABC can 
+be any 3 digits.`
+
+![Invalid param format](images/error-invalid-param-format.png)
+
+Typically this happens because the parameter provided does not match the required format. Check if your parameter conforms
+to the format stated under the [command parameters](#command-parameters) section. The error message usually gives some
+helpful information in resolving the error too.
+
+##### Commands that require an `INDEX`
+The bounds of the `INDEX` parameter is documented [below](#index). This section will explain how errors with the `INDEX`
+will be dealt with
+
+- If you do not specify the `INDEX` (e.g `rdel`): The app will inform you that the format is invalid, and display the 
+  format that should be used. This is a specific case of [incorrect formats](#incorrect-formats).
+  
+- If the `INDEX` is not an integer (e.g `rdel abc`): The app will inform you that the format is invalid, and display the
+  format that should be used. This is also a specific case of [incorrect formats](#incorrect-formats), as the command 
+  expects a number but did not get one.
+
+- If the `INDEX` parameter is required alongside other parameters e.g [`redit`](#edit-a-resident-record--redit), and
+  **only** an `INDEX` (be it valid or not) value is provided (e.g. `redit 1000` or `redit 1`): The app will inform you
+  that the format is invalid, and display the format that should be used. This is another specific case 
+  of [incorrect formats](#incorrect-formats), as the command compulsorily needs at least one of the optional parameters 
+  to be provided.
+  
+- If the `INDEX` is 0 or lesser (e.g `rdel -5`): The app will inform you that `INDEX` should be a positive integer.
+
+![Index non positive](images/error-idx-non-positive.png)
+
+- If the `INDEX` is above the range of the collection specified (e.g `rdel 100` but there are only 10 residents): The 
```

Comment:
> but there are only 10 residents

Maybe change this to

> but there are only 6 residents

to better match the image that follows.

Reply:
Done